### DISCUSSION TITLE
setInlineBlockType() command alternative in tiptap 2.0

### DISCUSSION THREAD

**Discussion by Gera814** [ueberdosis/tiptap#2798] [General Questions]
Hey there,

I am trying to find the **setInlineBlockType()** alternative in tiptap 2.0

As an alternative, I am trying to use the **setNode()** commad, but I got this warn:

`[tiptap warn]: Currently "setNode()" only supports text block nodes.`

**My tiptap 1 code:**
```
 commands({ type }) {
    return attrs => {
      return setInlineBlockType(type, getAttributes(attrs.kind))
    }
  }
```

**Comment by mattiastofte** [comment]
I'm having the same problem!

**Comment by mattiastofte** [comment]
I've found a solution, although it's not the prettiest. You can create a new inline NodeView node with [insertcontent](https://tiptap.dev/api/commands/insert-content)

```typescript
  addCommands() {
    return {
      insertLatexInline: attributes => ({ commands }) => {
        return commands.insertContent('<LatexInline></LatexInline>');
      },
    }
  },
```

If you'd like to also set the attributes, in my case I have a latex attribute (that corresponds to a string value containing a latex formula), you can do like so:

```typescript
  addCommands() {
    return {
      insertLatexInline: attributes => ({ commands }) => {
        const parameter = `$${attributes?.latexString}$`;
        return commands.insertContent(`<LatexInline latex=${parameter}></LatexInline>`);
      },
    }
  },
```

**Reply by pointnet** [reply]
You can use the nodespec create method to get rid of string interpolations, here I assume your `<LatexInline>` node name is `latexInline`:

```javascript
  addCommands() {
    return {
      insertLatexInline: attributes => ({ commands, editor }) => {
        const latexNode = editor.schema.nodes.latexInline.create({
          latex: attributes?.latexString ?? ""
        });
        return commands.insertContent(latexNode);
      },
    }
  },
```

**Reply by mattiastofte** [reply]
I agree with you, this solution is prettier. I'll try it out later tonight. Thanks!:)

**Comment by cnicodeme** [comment]
For inforamtion, I've replaced `setNode` by `wrapIn` and it works:

```javascript
addCommands () {
        return {
            setCallout: attributes => ({ commands }) => {
                return commands.wrapIn(this.type, attributes)
            }
        }
    }
```

**Reply by jelling** [reply]
Fwiw `wrapIn` isn't doing anything for me at all when trying to wrap the selected text inside a new node.
